TKGame/Debug.cs

```diff
@@ -150,12 +235,55 @@ private static void UpdateKeyboardOverlay()
             // For each key that is in the dictionary, check to see if it's
             // pressed. If it is, make it the active color, if it isn't, make it
             // the inactive color.
+            KeyboardState keyboardState = Keyboard.GetState();
+
             foreach (var keyLabelPair in keyboardLabelDict)
             {
-                keyLabelPair.Value.TextColor = (Keyboard.GetState().IsKeyDown(keyLabelPair.Key))
-                    ? DEBUG_KEYBOARD_OVERLAY_ACTIVE_COLOR
-                    : DEBUG_KEYBOARD_OVERLAY_INACTIVE_COLOR;
+                Label keyLabel = keyLabelPair.Value;
+
+                keyLabel.TextColor = (keyboardState.IsKeyDown(keyLabelPair.Key))
+                    ? KEYBOARD_OVERLAY_ACTIVE_TEXT_COLOR
+                    : KEYBOARD_OVERLAY_INACTIVE_TEXT_COLOR;
+
+                keyLabel.Background = (keyboardState.IsKeyDown(keyLabelPair.Key))
+                    ? ActiveBackgroundBrush
+                    : InactiveBackgroundBrush;
             }
         }
+
+        /// <summary>
+        /// Draw bounding rectangle around a given rectangle.
+        /// </summary>
+        public static void DrawBoundingBox(SpriteBatch spriteBatch, Rectangle rectangle, Color color, int lineWidth)
+        {
+            if (HitboxTexture is null)
+            {
+                HitboxTexture = new Texture2D(spriteBatch.GraphicsDevice, 1, 1);
+                HitboxTexture.SetData(new Color[] { Color.White });
+            }
+
+            // These will draw thin rectangles (essentially lines) on each edge of a rectangle
+            spriteBatch.Draw(HitboxTexture, new Rectangle(rectangle.X, rectangle.Y, lineWidth, rectangle.Height + lineWidth), color);
+            spriteBatch.Draw(HitboxTexture, new Rectangle(rectangle.X, rectangle.Y, rectangle.Width + lineWidth, lineWidth), color);
+            spriteBatch.Draw(HitboxTexture, new Rectangle(rectangle.X + rectangle.Width, rectangle.Y, lineWidth, rectangle.Height + lineWidth), color);
+            spriteBatch.Draw(HitboxTexture, new Rectangle(rectangle.X, rectangle.Y + rectangle.Height, rectangle.Width + lineWidth, lineWidth), color);
+        }
+
+        /// <summary>
+        /// Draw a bounding rectangle around a given entity.
+        /// </summary>
+        public static void DrawBoundingBox(SpriteBatch spriteBatch, Entity entity, Color color, int lineWidth)
+        {
+            // Construct a Rectangle out of the entity's position and size so we can
+            // just use the other DrawBoundingRectangle() function to actually draw
+            Rectangle entityRect = new Rectangle
+            {
```

Comment:
Do these need to be curly braces or should they be parentheses?

Reply:
Doesn't really matter as long as it works, just wondering since most of the time there are parentheses after new Rectangle.